node/src/components/block_synchronizer/block_acquisition.rs

```diff
@@ -1032,7 +1041,16 @@ impl BlockAcquisitionState {
                     "BlockAcquisition: registering execution result or chunk for: {}",
                     block.hash()
                 );
-                let deploy_hashes = block.deploy_and_transfer_hashes().copied().collect();
+                let deploy_hashes = match block.as_ref() {
+                    Block::V1(v1) => v1.deploy_and_transfer_hashes().copied().collect(),
+                    Block::V2(v2) => v2
+                        .all_transactions()
+                        .filter_map(|txn_hash| match txn_hash {
+                            TransactionHash::Deploy(deploy_hash) => Some(*deploy_hash),
+                            TransactionHash::V1(_) => None,
```

Comment:
Same question as above for `register_block`.

Reply:
As per comment above.